array.c

```diff
@@ -4692,198 +4692,251 @@ rb_ary_clear(VALUE ary)
  *
- *  If +start+ is negative, counts from the end (starting index is <tt>start + array.size</tt>):
+ *    a = ('a'..'d').to_a # => ["a", "b", "c", "d"]
+ *    a.fill('-')         # => ["-", "-", "-", "-"]
  *
- *    a = ['a', 'b', 'c', 'd']
- *    a.fill(:X, -2) # => ["a", "b", :X, :X]
+ *  With argument +object+ and numeric argument +start+,
+ *  replaces elements based on the given +start+:
  *
- *  If +start+ is too small (less than and far from zero), replaces all elements:
+ *  - If +start+ is in range (<tt>0 <= start < array.size</tt>),
+ *    replaces all elements from offset +start+ through the end:
  *
- *    a = ['a', 'b', 'c', 'd']
- *    a.fill(:X, -6) # => [:X, :X, :X, :X]
- *    a = ['a', 'b', 'c', 'd']
- *    a.fill(:X, -50) # => [:X, :X, :X, :X]
+ *      a = ('a'..'d').to_a # => ["a", "b", "c", "d"]
+ *      a.fill('-', 2)      # => ["a", "b", "-", "-"]
  *
- *  With arguments +obj+, Integer +start+, and Integer +length+, and no block given,
- *  replaces elements based on the given +start+ and +length+.
+ *   - If +start+ is too large (<tt>start >= array.size</tt>), does nothing:
  *
- *  If +start+ is in range, replaces +length+ elements beginning at offset +start+:
+ *      a = ('a'..'d').to_a # => ["a", "b", "c", "d"]
+ *      a.fill('-', 4)      # => ["a", "b", "c", "d"]
  *
- *    a = ['a', 'b', 'c', 'd']
- *    a.fill(:X, 1, 1) # => ["a", :X, "c", "d"]
+ *  - If +start+ is negative, counts from the end (starting index is <tt>start + array.size</tt>):
  *
- *  If +start+ is negative, counts from the end:
+ *      a = ('a'..'d').to_a # => ["a", "b", "c", "d"]
+ *      a.fill('-', -3)     # => ["a", "-", "-", "-"]
  *
- *    a = ['a', 'b', 'c', 'd']
- *    a.fill(:X, -2, 1) # => ["a", "b", :X, "d"]
+ *  - If +start+ is too small (less than and far from zero), replaces all elements:
  *
- *  If +start+ is large (<tt>start >= array.size</tt>), extends +self+ with +nil+:
+ *      a = ('a'..'d').to_a # => ["a", "b", "c", "d"]
+ *      a.fill('-', -9)     # => ["-", "-", "-", "-"]
  *
- *    a = ['a', 'b', 'c', 'd']
- *    a.fill(:X, 5, 0) # => ["a", "b", "c", "d", nil]
- *    a = ['a', 'b', 'c', 'd']
- *    a.fill(:X, 5, 2) # => ["a", "b", "c", "d", nil, :X, :X]
+ *  With argument +object+ and numeric arguments +start+ and +length+,
+ *  replaces elements based on the given +start+ and +length+:
  *
- *  If +length+ is zero or negative, replaces no elements:
+ *  - If +start+ is in range, replaces +length+ elements beginning at offset +start+:
  *
- *    a = ['a', 'b', 'c', 'd']
- *    a.fill(:X, 1, 0) # => ["a", "b", "c", "d"]
- *    a.fill(:X, 1, -1) # => ["a", "b", "c", "d"]
+ *      a = ('a'..'d').to_a # => ["a", "b", "c", "d"]
+ *      a.fill('-', 1, 2)   # => ["a", "-", "-", "d"]
  *
- *  With arguments +obj+ and Range +range+, and no block given,
- *  replaces elements based on the given range.
+ *  - If +start+ is negative, counts from the end:
```

Comment:
This is missing the case when `start <= -self.size`.

Reply:
Should be added.